go/arrow/datatype_nested.go

```diff
@@ -490,7 +495,12 @@ func (t *unionType) init(fields []Field, typeCodes []UnionTypeCode) {
 	}
 }
 
-func (t unionType) Fields() []Field            { return t.children }
+func (t unionType) Fields() []Field {
+	fields := make([]Field, len(t.children))
+	copy(fields, t.children)
+	return fields
+}
```

Comment:
same here. add a godoc comment to this to indicate it makes a copy of the fields

Reply:
1e81c723eb665da8f59519710c749e0fda435676